Automattic/jetpack
# Add WooCommerce address and profile settings to the v1.4 settings endpoint.

This PR adds the following "settings" to the `v1.4/sites/example.wordpress.com/settings` endpoint.

- (string) 'woocommerce_store_address'
- (string) 'woocommerce_store_address_2'
- (string) 'woocommerce_store_city'
- (string) 'woocommerce_default_country'
- (string) 'woocommerce_store_postcode'
- (array) 'woocommerce_onboarding_profile'

On a `GET` request, the address part settings return the user's associated domain information address, if it's set. Otherwise they return an empty string.

On a `POST` request the address part fields are sanitized using `sanitize_text_field()`.

`woocommerce_onboarding_profile` is a multi-dimensional array and may require some complex sanitation. TBD.

#### Jetpack product discussion
p9o2xV-1L5-p2

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:

Run unit tests with the command:
`jetpack docker phpunit -- --filter=WP_Test_WPCOM_JSON_API_Site_Settings_V1_4_Endpoint`

Using the Wordpress.com developer API console:

The console only goes to version 1.3, but you can run a GET|POST request to /sites/example.wordpress.com/settings then "Edit and Resend" using your browser's Network web tools. From there you can run a POST request with a custom body and change the endpoint to 1.4.

Test saving various values to each field using POST and confirm they were set with the GET request.

POST body examples:
`{"woocommerce_onboarding_profile":["Test", "One", "Two"]}`
`{"woocommerce_store_address":"Test Address 123"}`
